donate/payments/tasks.py

```diff
@@ -245,6 +252,82 @@ def process_charge_succeeded(self, event):
             transaction_fee,
         )
 
+        if settings.MIGRATE_STRIPE_SUBSCRIPTIONS_ENABLED and 'thunderbird' not in metadata:
+            MigrateStripeSubscription().process(charge, subscription)
+
+
+class MigrateStripeSubscription:
+    def process(self, charge, subscription):
+        last4 = subscription.customer.sources.data[0].last4
+
+        # Fetch the Customer object that was migrated to Braintree from Stripe
+        # along with associated payment information
+        BT_customers = gateway.customer.search(
+            braintree.TransactionSearch.customer_email == charge.customer_email,
+            braintree.TransactionSearch.credit_card_number.ends_with(last4)
+        )
+
+        # Ensure we found a customer record, if not, report it to Sentry as an error
+        if len(BT_customers) == 0:
+            logger.error(f'No Customer record exists in Braintree '
+                         f'for the Stripe customer attached to this charge object:'
+                         f'{charge.id}', exc_info=True)
+            return
+
+        # There should be a single Customer/Card combo on Braintree
+        BT_customer = BT_customers[0]
+
+        # Ensure there is a PaymentMethod associated with the customer.
+        # If not, report the error to Sentry
+        if len(BT_customer.payment_methods) == 0:
+            logger.error(f'The Braintree customer ({BT_customer.id}) does not have '
+                         f'an associated PaymentMethod', exc_info=True)
+            return
+
+        # Grab the Payment Method token that represents the customers payment information in the Braintree Vault
+        payment_method_token = BT_customer.payment_methods[0].token
+
+        # Determine the Braintree plan, merchant account, and price for the subscription
+        # based on the Stripe subscription
+        BT_plan_id = get_plan_id(subscription.plan.currency)
+        BT_merchant_account = get_merchant_account_id_for_card(subscription.plan.currency)
+        BT_price = zero_decimal_currency_fix(charge.amount, subscription.plan.currency)
```

Comment:
Note: we 100% want to validate that zero-decimal currency subscriptions properly migrate.